### PR TITLE
Use __context__ to optimize lgpo module

### PR DESCRIPTION
**Pull request by twangboy**
### What does this PR do?
Uses __context__ to store information that was being generated repeatedly. This should help with the speed issues.

@lomeroe Could you give this a once-over in case I broke something?

### Tests written?
No

### Commits signed with GPG?
Yes

### CODE REVIEW

**Comment by rallytime** [comment]
Hey @twangboy Can you fixup this lint error?

https://jenkinsci.saltstack.com/job/pr-lint/job/PR-49919/2/

**Comment by cachedout** [comment]
I don't see *any* automated testing for LGPO. Is that correct? If so, this is an awful big change to make without a safety net.

File: salt/modules/win_lgpo.py [hunk 1]
```diff
@@ -3555,72 +3572,80 @@ def _buildElementNsmap(using_elements):
...
+    '''
+    # Set file names
+    f_sdb = os.path.join(__opts__['cachedir'], 'secedit-{0}.sdb'.format(UUID))
+    f_inf = os.path.join(__opts__['cachedir'], 'secedit-{0}.inf'.format(UUID))
+
+    # Write the changes to the inf file
+    __salt__['file.write'](f_inf, inf_data)
+
+    # Run secedit to make the change
+    retcode = __salt__['cmd.retcode'](['secedit', '/configure', '/db', f_sdb,
```
Comment (dwoz) [inline salt/modules/win_lgpo.py]: Should this be in a try/finally to make sure we do not leave stale files around?

**Review by dwoz — CHANGES_REQUESTED** [review]

**Comment by lomeroe** [comment]
@twangboy I haven't been able to really look at this, but I *tried* to add a module test in #50006

**Comment by cachedout** [comment]
@dwoz Can you please take another swing back through here?

**Comment by lomeroe** [comment]
@twangboy I don't see anything glaring when looking through it.  Throwing the test added in #50006 at it, the only failures are those that are fixed in that PR, so *seems* okay.

**Review by dwoz — APPROVED** [review]

**Pull request merged**